apps/tx-builder/src/utils.test.ts

```diff
@@ -0,0 +1,1506 @@
+        const tupleWithNestedTuplesFieldType = getInputTypeHelper({
+          name: 's',
+          type: 'tuple',
+          internalType: 'tuple',
+          components: [
+            {
+              name: 'a',
+              internalType: 'uint256',
+              type: 'uint256',
+            },
+            {
+              name: 'b',
+              internalType: 'uint256[]',
+              type: 'uint256[]',
+            },
+            {
+              name: 'c',
+              type: 'tuple[]',
+              internalType: 'tuple[]',
+              components: [
+                {
+                  name: 'x',
+                  internalType: 'uint256',
+                  type: 'uint256',
+                },
+                {
+                  name: 'y',
+                  internalType: 'uint256',
+                  type: 'uint256',
+                },
+                {
+                  name: 'z',
+                  internalType: 'tuple',
+                  type: 'tuple',
+                  components: [
+                    {
+                      name: 'a',
+                      internalType: 'uint256',
+                      type: 'uint256',
+                    },
+                    {
+                      name: 'b',
+                      internalType: 'uint256',
+                      type: 'uint256',
+                    },
+                  ],
+                },
+              ],
+            },
+          ],
+        })
+
+        // tuple(uint256,uint256[],tuple(uint256,uint256,tuple(uint256,uint256))[])
+
+        expect(
+          parseInputValue(tupleWithNestedTuplesFieldType, '[1,[2,3],[[3,3,[5,5]],[4,4,[6,6]]]]'),
+        ).toEqual([
+          1,
+          [2, 3],
+          [
+            [3, 3, [5, 5]],
+            [4, 4, [6, 6]],
+          ],
+        ])
+      })
+
+      it.skip('tuples with long int values are parsed to valid BN values', () => {
+        const tupleFieldType = getInputTypeHelper({
+          components: [
+            {
+              internalType: 'uint128',
+              name: 'allocated',
+              type: 'uint128',
+            },
+            {
+              internalType: 'uint128',
+              name: 'loss',
+              type: 'uint128',
+            },
+          ],
+          internalType: 'struct AllocatorLimits',
+          name: 'limits',
+          type: 'tuple',
+        })
+
+        // tupleFieldType is tuple(uint128,uint128)
+        expect(
+          parseInputValue(tupleFieldType, '["6426191757410075707","6426191757410075707"]'),
+        ).toEqual(['6426191757410075707', '6426191757410075707'])
+        // FIX: fix the issue with long numbers in tuples
+        expect(
+          parseInputValue(tupleFieldType, '[6426191757410075707,6426191757410075707]'),
+        ).toEqual(['6426191757410075707', '6426191757410075707'])
+      })
+
+      it.skip('parses a tuple with nested tuples with long int values', () => {
```

Comment:
I skipped this test because the issue with long numbers in tuples will be fixed in this other ticket: https://github.com/safe-global/safe-react-apps/issues/448

I added this comment below:
```
        // FIX: fix the issue with long numbers in tuples
```